cpp/src/arrow/csv/writer_test.cc

```diff
@@ -279,7 +290,20 @@ std::vector<WriterTestParams> GenerateTestCases() {
       {schema_custom_delimiter, batch_custom_delimiter,
        DefaultTestOptions(/*include_header=*/false, /*null_string=*/"",
                           QuotingStyle::Needed, /*eol=*/";", /*delimiter=*/';'),
-       /*expected_output*/ "", expected_status_illegal_delimiter(';')}};
+       /*expected_output*/ "", expected_status_illegal_delimiter(';')},
```

Comment:
It seems that all three `QuotingStyle` values should be tested with a mix of column names requiring quoting and column names that don't need quoting.